### DISCUSSION TITLE
How to use the "editorconfig" in a .prettierrc in v3.0.3

### DISCUSSION THREAD

**Discussion by ChristianSchwarz** [prettier/prettier#15603] [Q&A] [answered]
I have prettier v3.0.3 and this `.prettierrc` config:

```
{
  "editorconfig": true,
  "bracketSpacing": true,
  "htmlWhitespaceSensitivity": "css",
  "proseWrap": "preserve",
  "quoteProps": "as-needed",
  "semi": true,
  "trailingComma": "none",
  "arrowParens": "avoid",
  "singleAttributePerLine": true
}
```

But when I run "prettier:fix" I get this warning: 
`[warn] Ignored unknown option { editorconfig: true }.`

Whats wrong here? 
Also I wonder where prettier looks for the `.editorconfig` file. In the same folder where the .prettierrc is, or also in parent folders?

StackOverflow: https://stackoverflow.com/questions/77436681/how-to-use-the-editorconfig-in-a-prettierrc-in-v3-0-3

**Comment by dudeofawesome** [comment]
I suspect that you might be getting confused by the prettier API's option for `prettier.resolveConfig` as I was.

https://prettier.io/docs/en/api.html#prettierresolveconfigfileurlorpath--options

**Comment by sosukesuzuki** [comment, marked as answer]
You cannot use the `editorconfig` option inside `.prettierrc`. Prettier use `.editorconfig` as the configuration file if `.prettierrc` does not exist and a `.editorconfig` file exists.

**Reply by pqnet** [reply]
There is a confusing statement in the documentation here:
https://prettier.io/docs/en/configuration#editorconfig:

> If options.editorconfig is true and an [.editorconfig file](https://editorconfig.org/) is in your project, Prettier will parse it and convert its properties to the corresponding Prettier configuration. This configuration will be overridden by .prettierrc, etc.

This suggests that, differently from what stated here:
- there is a `editorconfig` option to enable/disable this behavior (this comment suggests the behavior is enabled by default and not configurable - which seems to be what actually happens)
- prettier will merge settings from `.editorconfig` and `.prettierrc`, giving priority to `.prettierrc` if they are different (this comment suggest that `.editorconfig` file is ignored if `.prettierrc` is present)

Can we get some clarity on this?
